test/lib/skunk/application_test.rb

```diff
@@ -27,5 +28,26 @@
         _(result).must_equal success_code
       end
     end
+
+    context "when passing --out option with a file" do
+      require "fileutils"
+
+      let(:argv) { ["--out=tmp/generated_report.txt", "samples/rubycritic"] }
+      let(:success_code) { 0 }
+
+      it "writes output to the file" do
+        FileUtils.rm("tmp/generated_report.txt", force: true)
+        FileUtils.mkdir_p("tmp")
+
+        RubyCritic::AnalysedModule.stub_any_instance(:churn, 1) do
```

Comment:
@etagwerker I added these stubs to address your comments about testing the generated content in the files. Note I applied these at the level of the application that is where the file is output, not in the argv parser as you suggested.